platform/InternalIOErr.roc

```diff
@@ -0,0 +1,58 @@
+module [
+    IOErr,
+    IOErrFromHost,
+    handle_err,
+]
+
+## **NotFound** - An entity was not found, often a file.
+##
+## **PermissionDenied** - The operation lacked the necessary privileges to complete.
+##
+## **BrokenPipe** - The operation failed because a pipe was closed.
+##
+## **AlreadyExists** - An entity already exists, often a file.
+##
+## **Interrupted** - This operation was interrupted. Interrupted operations can typically be retried.
+##
+## **Unsupported** - This operation is unsupported on this platform. This means that the operation can never succeed.
+##
+## **OutOfMemory** - An operation could not be completed, because it failed to allocate enough memory.
+##
+## **Other** - A custom error that does not fall under any other I/O error kind.
+IOErr : [
+    NotFound,
+    PermissionDenied,
+    BrokenPipe,
+    AlreadyExists,
+    Interrupted,
+    Unsupported,
+    OutOfMemory,
+    Other Str,
+]
+
+IOErrFromHost : {
+    tag : [
+        EndOfFile,
+        NotFound,
+        PermissionDenied,
+        BrokenPipe,
+        AlreadyExists,
+        Interrupted,
+        Unsupported,
+        OutOfMemory,
+        Other,
+    ],
+    msg : Str,
+}
+
+handle_err : IOErrFromHost -> IOErr
+handle_err = \{ tag, msg } ->
+    when tag is
+        NotFound -> NotFound
+        PermissionDenied -> PermissionDenied
+        BrokenPipe -> BrokenPipe
+        AlreadyExists -> AlreadyExists
+        Interrupted -> Interrupted
+        Unsupported -> Unsupported
+        OutOfMemory -> OutOfMemory
+        Other | EndOfFile -> Other msg
```

Comment:
From an [earlier PR](https://github.com/roc-lang/basic-cli/pull/288):

> I know this was already there, but this maps everything one-to-one except for EndOfFile. Since this method is generically called handle_err, this seems like a bad idea, but maybe there's a good reason.

Reply:
Yes, the reason is that the `Err` tag used in the Host version is that it's needed for the Stdin API...

```roc
handle_err : InternalIOErr.IOErrFromHost -> [EndOfFile, StdinErr Err]
```

This file was copied verbatim from basic-cli and so I have left this here, so that this version is the same. The intention is to move all these types into a common package where we can, but this is a future design discussion that we haven't resolved.